modules/core/keeper/msg_server.go

```diff
@@ -697,3 +698,16 @@ func (k Keeper) Acknowledgement(goCtx context.Context, msg *channeltypes.MsgAckn
 
 	return &channeltypes.MsgAcknowledgementResponse{Result: channeltypes.SUCCESS}, nil
 }
+
+// UpdateClientParams defines a rpc handler method for MsgUpdateClientParams.
+func (k Keeper) UpdateClientParams(goCtx context.Context, msg *clienttypes.MsgUpdateClientParams) (*clienttypes.MsgUpdateClientParamsResponse, error) {
+	ck := k.ClientKeeper
+	if ck.GetAuthority() != msg.Authority {
```

Comment:
since the entrypoint(s) to the msg server for ibc core submodules is in the core keeper, it might make sense for `GetAuthority` to just exist at the level of the core ibc keeper rather than passing the authority along to each nested keeper (client, connection). 

I don't see a world right now in which client would have authority x and connection would have authority y.

Thoughts? cc. @colin-axner

Reply:
that's seems like a good point to me. I'll wait for Colin's thoughts on this before making changes.

Reply:
I've made this change. Code definitely looks better. Will revert if required.